packages/server/src/main/java/software/uncharted/terarium/hmiserver/repository/notification/NotificationGroupRepository.java

```diff
@@ -37,4 +37,30 @@ List<NotificationGroup> findAllUnackedByUserReturnAllEvents(
 		@Param("userId") @NotNull String userId,
 		@Param("createdOn") @NotNull Timestamp createdOn
 	);
+
+	@Query(
+		"""
+		SELECT ne
+			FROM NotificationEvent ne
+			WHERE ne.createdOn = (
+					SELECT MAX(ne2.createdOn)
+					FROM NotificationEvent ne2
+					WHERE ne2.notificationGroup.id = ne.notificationGroup.id
+			)
+			AND ne.notificationGroup.userId = :userId
+			 AND ne.notificationGroup.createdOn > :createdOn
+			AND EXISTS (
+					SELECT 1
+					FROM NotificationEvent ne3
+					WHERE ne3.notificationGroup.id = ne.notificationGroup.id
+					AND ne3.acknowledgedOn IS NULL
+			)
+			ORDER BY ne.notificationGroup.createdOn DESC
+
+		"""
+	)
+	List<NotificationGroup> findAllUnackedByUserReturnAllEventsLatestOnly(
+		@Param("userId") @NotNull String userId,
```

Comment:
Is this repository method being used in anywhere? I can't seem to find the place this is used in.

Reply:
Sorry, let me remove it. It doesn't work because hibernate always pulls all join fields...